modules/programs/thunderbird.nix

```diff
@@ -0,0 +1,269 @@
+              type = with types; listOf str;
+              default = [ ];
+              example = literalExpression ''
+                [ "profile1" "profile2" ]
+              '';
+              description = ''
+                List of Thunderbird profiles for which this account should be
+                enabled. If this list is empty (the default), this account will
+                be enabled for all declared profiles.
+              '';
+            };
+
+            settings = mkOption {
+              type = with types; functionTo (attrsOf (oneOf [ bool int str ]));
+              default = _: { };
+              defaultText = literalExpression "_: { }";
+              example = literalExpression ''
+                id: {
+                  "mail.identity.id''${id}.protectSubject" = false;
+                  "mail.identity.id''${id}.autoEncryptDrafts" = false;
+                };
+              '';
+              description = ''
+                Extra settings to add to this Thunderbird account configuration.
+                The <varname>id</varname> given as argument is an automatically
+                generated account identifier.
+              '';
+            };
+          };
+        });
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      (hm.assertions.assertPlatform "programs.thunderbird" pkgs platforms.linux)
+
+      (let defaults = catAttrs "name" (filter (a: a.isDefault) profilesWithId);
+      in {
+        assertion = cfg.profiles == { } || length defaults == 1;
+        message = "Must have exactly one default Thunderbird profile but found "
+          + toString (length defaults) + optionalString (length defaults > 1)
+          (", namely " concatStringsSep "," defaults);
+      })
+
+      (let
+        profiles = catAttrs "name" profilesWithId;
+        selectedProfiles =
+          concatMap (a: a.thunderbird.profiles) enabledAccounts;
+      in {
+        assertion = (intersectLists profiles selectedProfiles)
+          == selectedProfiles;
+        message = "Cannot enable an account for a non-declared profile. "
+          + "The declared profiles are " + (concatStringsSep "," profiles)
+          + ", but the used profiles are "
+          + (concatStringsSep "," selectedProfiles);
+      })
+    ];
+
+    home.packages = [ cfg.package ]
+      ++ optional cfg.withExternalGnupg pkgs.gpgme;
+
+    home.file = mkMerge ([{
+      "${thunderbirdConfigPath}/profiles.ini" =
+        mkIf (cfg.profiles != { }) { text = generators.toINI { } profilesIni; };
+    }] ++ flip mapAttrsToList cfg.profiles (name: profile: {
+      "${thunderbirdConfigPath}/${name}/user.js" = let
+        accounts = filter (a:
+          a.thunderbird.profiles == [ ]
+          || any (p: p == name) a.thunderbird.profiles) enabledAccountsWithId;
+      in mkIf (cfg.settings != { } || length accounts != 0) {
+        text = concatStringsSep "\n" (map mkUserJs ([
+          cfg.settings
+
+          profile.settings
```

Comment:
imo here profile.settings should override cfg.settings, maybe `cfg.settings  // profile.settings` even, if later options override earlier options in the format.